### DISCUSSION TITLE
A Cell That Prevents Killing Similar Organisms

### DISCUSSION THREAD

**Discussion by theomgdev** [MaxRobinsonTheGreat/EvolutionSimulatorV2#15] [Ideas and Suggestions]
If they had the option of not killing the same or similar organisms, I'm sure the organisms would show organized behavior and a tendency to group. With eyes, the brain also can carry a "kill the stranger" or "follow the relative" attitude. For similarity measure, you can convert organism data to a plain string and use Levenshtein distance.

**Comment by PhoneixS** [comment]
This could be a new cell type "Id cell" which simply stores a number.  However, the id cell is different than other cells in that if they are seen by an eye they could have two behaviors (instead of one) depending on if the number is similar to one of themself or they don't share any id cell.

I think this could make evolving creatures that have the same behaviors as with the "cell that prevents killing" but at the same time give opportunities to emerging other behaviors like disguise as "herbivorous".

Maybe you can implement it as a score or percent instead of simply being it has the same or not.

**Comment by MaxRobinsonTheGreat** [comment]
I think the best way to do this is to selectively kill organisms from a different species. Organisms know their species, which only changes when they mutate cells. This can be used to decide if this "Selective killer" cell will harm touched organisms. This would add no additional computational complexity.
I plan to try this out when I have time, very curious how this will affect evolution.

**Comment by SelfSealShelf** [comment]
They can and do evolve this trait on their own. It's extremely advantageous, but rare.

**Reply by MaxRobinsonTheGreat** [reply]
They can't do this, they can approximate it by avoiding certain types of cells which is not the same thing, and killer cells harm indiscriminately. It's also not necessarily advantageous, which is why I think it would be an interesting experiment.
